# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxPlatformMac.cpp
# Commit: a01eaeed432a
# Full Hash: a01eaeed432ab996a3f5275229be89814e92b95a
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-05-18 21:36:28
# Regressor Bug: 1708768
# File Overlap Count: 2
# Description:
#   Bug 1708768 - Preliminary cleanup of gfxPlatformFontList creation/initialization code; no functional change. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D114863
# ==============================================================================

diff -r 39c32384d7a9 -r a01eaeed432a gfx/thebes/gfxPlatformMac.cpp
--- a/gfx/thebes/gfxPlatformMac.cpp	Tue May 18 10:47:43 2021 +0000
+++ b/gfx/thebes/gfxPlatformMac.cpp	Tue May 18 11:10:42 2021 +0000
@@ -238,13 +238,8 @@
 
 bool gfxPlatformMac::ContentUsesTiling() const { return UsesTiling(); }
 
-gfxPlatformFontList* gfxPlatformMac::CreatePlatformFontList() {
-  gfxPlatformFontList* list = new gfxMacPlatformFontList();
-  if (NS_SUCCEEDED(list->InitFontList())) {
-    return list;
-  }
-  gfxPlatformFontList::Shutdown();
-  return nullptr;
+bool gfxPlatformMac::CreatePlatformFontList() {
+  return gfxPlatformFontList::Initialize(new gfxMacPlatformFontList);
 }
 
 void gfxPlatformMac::ReadSystemFontList(SystemFontList* aFontList) {